MiniTwitch.Common/WebSocketClient.cs

```diff
@@ -204,15 +198,15 @@ public async ValueTask SendAsync(string data, bool sensitive = false, Cancellati
 
         try
         {
-            await _client!.SendAsync(mem.Memory[..written], WebSocketMessageType.Text, true, cancellationToken);
+            await _client!.SendAsync(new ReadOnlyMemory<byte>(bytes)[..written], WebSocketMessageType.Text, true, cancellationToken);
         }
         catch (Exception ex)
         {
             LogException(ex, "Exception caught whilst trying to send data.");
         }
         finally
         {
-            mem.Dispose();
+            ArrayPool<byte>.Shared.Return(bytes);
```

Comment:
Since this is library code, would it make sense for each instance of this class to own its own pool? If you're dipping into shared array pools, you could be touching the application code's array pools.

Reply:
If you use https://learn.microsoft.com/en-us/dotnet/api/system.buffers.arraypool-1.create?view=net-7.0#system-buffers-arraypool-1-create(system-int32-system-int32) you'll also have a bit more control over sizing/initial capacity

Reply:
I like this idea, but unsure where and how accessible my own pool should be. Also, you mention control over initial capacity, but I don't see it mentioned anywhere on the docs

Reply:
Sorry, I half assed that comment :D I meant bucket size not initial size. Your pool should be internal